web-client/public/js/generateNetwork.js

```diff
@@ -94,34 +104,58 @@ export const generateNetwork = function () {
-                    grnState.customWorkbook.genes[response.geneId] = response.displayGeneId;
-                    displayCurrentGenes();
-                } else {
-                    alert(
-                        `Gene: ${searchGene} was not found in this database. Please check for any typos and try again.`
-                    );
-                }
-            }).catch(function (error) {
-                console.log(error.stack);
-                console.log(error.name);
-                console.log(error.message);
-            });
+            if (grnState.customWorkbook.type === "grn") {
+                let headers = {
+                    type:"NetworkGeneFromSource",
+                    gene: geneProtein,
+                    source:grnState.customWorkbook.sources.geneRegulation[source].source,
+                    timestamp:grnState.customWorkbook.sources.geneRegulation[source].timestamp
+                };
+                queryNetworkDatabase(headers).then(function (response) {
+                    if (response.geneId && response.displayGeneId ) {
+                        grnState.customWorkbook.genes[response.geneId] = response.displayGeneId;
+                        displayCurrentGenes();
+                    } else {
+                        alert(
+                            `${userGeneProtein}: "${searchGeneProtein}" was not found in this database. Please check for any typos and try again.`
+                        );
+                    }
+                }).catch(function (error) {
+                    console.log(error.stack);
+                    console.log(error.name);
+                    console.log(error.message);
+                });
+            } else if (grnState.customWorkbook.type === "protein-protein-physical-interaction") {
+                let headers = {
+                    type:"NetworkFromGeneProtein",
+                    geneProtein: geneProtein,
+                    source:grnState.customWorkbook.sources.proteinProteinInteractions[source].source,
+                    timestamp:grnState.customWorkbook.sources.proteinProteinInteractions[source].timestamp
+                };
+                queryProteinProteinDatabase(headers).then(function (response) {
+                    if (response.standardName && response.displayGeneId && response.geneId) {
+                        grnState.customWorkbook.genes[response.standardName] = {displayGeneID: response.displayGeneId, geneID: response.geneId};
+                        displayCurrentGenes();
+                    } else {
+                        alert(
+                            `${userGeneProtein}: "${searchGeneProtein}" was not found in this database. Please check for any typos and try again.`
+                        );
+                    }
+                }).catch(function (error) {
+                    console.log(error.stack);
+                    console.log(error.name);
+                    console.log(error.message);
```

Comment:
Mental note for future work: we should probably place this information in our standard error/warning modal right?